### PR TITLE
[BufferGeometry] normalizeNormals

### PR DESCRIPTION
**Pull request by Neesnu**
Fix for InterleavedBufferAttribute normalizing a normal across an interleavedBuffer

### CODE REVIEW

File: src/core/BufferGeometry.js [hunk 1]
```diff
@@ -814,21 +814,19 @@ Object.assign( BufferGeometry.prototype, EventDispatcher.prototype, {
...
 
-		for ( var i = 0, il = normals.length; i < il; i += 3 ) {
+		for ( var i = 0, il = normals.count; i < il; i ++ ) {
 
-			x = normals[ i ];
-			y = normals[ i + 1 ];
-			z = normals[ i + 2 ];
+			x = normals.getX( i );
+			y = normals.getY( i );;
+			z = normals.getZ( i );;
```
Comment (WestLangley) [inline src/core/BufferGeometry.js]: double ;;

File: src/core/BufferGeometry.js [hunk 2]
```diff
@@ -814,21 +814,19 @@ Object.assign( BufferGeometry.prototype, EventDispatcher.prototype, {
...
 
 			n = 1.0 / Math.sqrt( x * x + y * y + z * z );
 
-			normals[ i ] *= n;
-			normals[ i + 1 ] *= n;
-			normals[ i + 2 ] *= n;
+			normals.setXYZ(i, normals.getX( i ) * n, normals.getY( i ) * n, normals.getZ( i ) * n)
```
Comment (WestLangley) [inline src/core/BufferGeometry.js]: normals.setXYZ( i, x * n, y * n, z * n );

File: src/core/BufferGeometry.js [hunk 3]
```diff
@@ -814,21 +814,19 @@ Object.assign( BufferGeometry.prototype, EventDispatcher.prototype, {
...
+		for ( var i = 0, il = normals.count; i < il; i ++ ) {
 
-			x = normals[ i ];
-			y = normals[ i + 1 ];
-			z = normals[ i + 2 ];
+			x = normals.getX( i );
+			y = normals.getY( i );;
+			z = normals.getZ( i );;
 
 			n = 1.0 / Math.sqrt( x * x + y * y + z * z );
```
Comment (Neesnu) [inline src/core/BufferGeometry.js]: Isn't this supposed to be n = 1.0 / ( x * x + y * y + z * z );
@mrdoob @WestLangley ?

**Comment by Neesnu** [comment]
Actually, I am losing my mind with a problem in my own code base, normal calculation is correct.

**Comment by Mugen87** [comment]
It looks like `normalizeNormals` only works for non-indexed geometries right now. I think we should also regard indexed geometries...

**Comment by mrdoob** [comment]
Thanks!

**Comment by Neesnu** [comment]
I am confused how normalizeNormals would change with respect to indexed geometries(as normals are bound to a vertex, not a triangle), although I can see how creating the normals would change with this respect, but I am pretty sure that is already covered.
I am going to add updating computeVertexNormals to my personal list of things to improve to support interleaved buffers - (make them not directly access the FloatArray).

**Comment by Mugen87** [comment]
Your're right. Sry, i've confused two things.

**Pull request merged**